### PR TITLE
feat(api-client): settings feature

### PR DESCRIPTION
**Pull request by DemonHa**
**Checklist**

I've gone through the following:

- [x] I've added an explanation _why_ this change is needed.
- [x] I've added a changeset (`pnpm changeset`).
- [x] I've added tests for the regression or new feature.
- [ ] I've updated the documentation.

---

> [!NOTE]
> Introduces a new settings UI (proxy selection, theme switching, appearance modes), adds exhaustive tests, tweaks sidebar responsiveness, and adjusts env vars table placeholder row.
> 
> - **Features (api-client v2)**:
>   - **Settings UI**: New `Settings.vue` with sections for `CORS Proxy`, `Themes`, `Framework Themes`, and `Appearance` in `packages/api-client/src/v2/features/settings/`.
>     - Components: `components/Appearance.vue`, `components/Section.vue`.
>     - Helper: `helpers/get-theme-colors.ts`.
>     - Export: `features/settings/index.ts` re-exports `Settings`.
>     - Tests: Comprehensive `Settings.test.ts` covering proxy, themes, integration themes, appearance, links, and edge cases.
> - **UI/UX**:
>   - Sidebar: Replace breakpoint-based inline style with utility class `hidden lg:block` in `components/sidebar/Sidebar.vue`.
> - **Environments**:
>   - Environment variables table: Placeholder row now includes `{ name: '', value: '', domain: '', isDisabled: true }` in `features/environments/components/EnvironmentVariablesTable.vue`.
> - **Release**:
>   - Changeset: Minor bump for `@scalar/api-client` with "feat: settings feature".
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 3eebb5a42b347dcde79b2cc8f001e74be41f36c4. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by DemonHa** [comment]
waiting for #7067

File: packages/api-client/src/v2/features/settings/Settings.vue [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+                buttonStyles({
+                  active: proxyUrl === customProxyUrl,
+                }),
+              )
+            "
+            @click="emit('update:proxyUrl', customProxyUrl)">
+            <div
+              class="bg-c-accent text-b-1 flex h-5 w-5 items-center justify-center rounded-full border-[1.5px] border-transparent p-1">
+              <ScalarIcon
+                v-if="true"
```
Comment (hanspagel) [inline packages/api-client/src/v2/features/settings/Settings.vue]: what cursor says

File: packages/api-client/src/v2/components/sidebar/Sidebar.vue [hunk 2]
```diff
@@ -24,14 +23,10 @@ defineSlots<{
...
 }>()
-
-const { breakpoints } = useBreakpoints()
 </script>
 <template>
   <Resize
-    :style="{
-      display: breakpoints.lg ? 'block' : 'none',
-    }"
+    class="hidden lg:block"
```
Comment (hanspagel) [inline packages/api-client/src/v2/components/sidebar/Sidebar.vue:29]: nice

**Review by geoffgscott — APPROVED** [review]
